tests/spec/s-array.js

```diff
@@ -1596,4 +1596,385 @@ describe('Array', function () {
             expect(result).toEqual(['2', '3']);
         });
     });
+
+    describe('#sort()', function () {
+        describe('usage', function () {
+            it('requires a function or undefined as first argument', function () {
+                var actual = [];
+                expect(actual.sort()).toBe(actual);
+                expect(actual.sort(undefined)).toBe(actual);
+
+                // http://ecma-international.org/ecma-262/5.1/#sec-15.4.4.11
+                // If comparefn is not undefined and is not a consistent comparison
+                // function for the elements of this array (see below), the behaviour
+                // of sort is implementation-defined.
+                // Some consitency would good here. Throw or not to throw.
+                expect(function () {
+                    actual.sort(null);
+                }).not.toThrow();
```

Comment:
instead of asserting these shouldn't throw, can instead they be `expect(actual.sort(null)).toBe(actual)` like the undefined tests above?